generator-theia/src/extension/extension-package-generator.ts

```diff
@@ -101,4 +104,26 @@ export class ExtensionPackageGenerator extends AbstractGenerator {
         }
     }
 
-}
\ No newline at end of file
+    protected compilePackageTdOptions(): object {
+        return {
+            "excludeNotExported": true,
+            "exclude": "lib, node_modules",
+            "experimentalDecorators": true,
+            "ignoreCompilerErrors": true,
+            "mode": "file",
+            "out": "./docs/" + this.getPackageName(),
+            "baseUrl": ".",
+            "paths": {
+                "@theia/*": ["node_modules/@theia/*"]
+            }
+        }
+    };
+
+    private getPackageName(): string {
+        let directoryName = `${process.cwd()}`;
+
+        const split = directoryName.split("/");
+        return split.reverse()[0];
```

Comment:
You should use https://nodejs.org/dist/latest-v8.x/docs/api/path.html to get that directory filename using / won't work on windows